### DISCUSSION TITLE
How to use negative values when using CSS Custom Properties in our spacings ?

### DISCUSSION THREAD

**Discussion by lauthieb** [tailwindlabs/tailwindcss#7247] [Help]
Hi everyone,

First of all, thank you for TailwindCSS, it's pretty awesome and we are proud to use it at Decathlon since June 2020 on https://github.com/decathlon/vitamin-web 😃 

I have a request (thanks to @bastienterrier), how to use negative values when we use CSS Custom Properties in our spacings?

Here is our Tailwind configuration: 
https://github.com/Decathlon/vitamin-web/blob/ba6de042a3fa473dd89ce544299f13dff4ee4fae/packages/sources/css/presets/tailwind/theme.js#L136-L147

As you can see, we use CSS Custom Properties in order to manage the source of truth for our design tokens in one place (for users that doesn't need Tailwind for example), here: https://github.com/Decathlon/vitamin-web/blob/main/packages/sources/css/src/design-tokens/src/index.css

The problem is that when we want to use negative values like `-mt-8` class, the CSS outputted is:
```css
.-mt-8 {
  margin-top: var(--spacing-8);
}
```

How can we handle this problem, please? 

Thanks a lot!

Laurent

**Comment by tpinne** [comment]
We have the same/similar problem as we use `clamp()` for our spacing values. I guess this won't be possible as they convert simple static positive values during compile time to simple static negative values (I guess).
Creating negative versions for dynamic values like variables or functions would only be possible with `calc`. And that might be hard to generalize. But we hope 🙏🏼
